### DISCUSSION TITLE
Store API keys in application, not browser?

### DISCUSSION THREAD

**Discussion by joergRossdeutscher** [opoto/wtracks#41] [Q&A]
Hi,

Thanks for this great web application. I am using wtracks protected by a base auth per .htaccess. I access that installation with multiple browsers on multiple devices.

It would be great if the API keys could be stored on the server instead of each single browser. Is that possible somehow?

Thanks in advance,
Jörg

**Comment by opoto** [comment]
Hi
WTracks is "serverless", no code is executed on the server, which only provides html/JavaScript processed in your browser. This is why your configuration has to be in your browser.
However WTracks supports exporting & importing your local configuration (including api keys, and map settings) to a file. What you could do is export to a file in a synchronized folder (eg with Dropbox, G.Drive, OneDrive, etc.), and import it to the other browsers from this shared storage.
Hope this helps.

**Comment by joergRossdeutscher** [comment]
Hello, thanks for your answer.

I totally understand that, however, since I definitely wanted that feature for me personally, I decided to go with a very ugly works-for-me quickhack. I didn't want to alter files in the repo, so git pull keeps working in the future. Instead I enforced a redirect on the config, so I could prepend it with mine:

.htaccess:
```
RewriteEngine On
RewriteRule js/config.js https://%{HTTP_HOST}/wtracks/js/config.php [R=307,L]

```
…and…

config.php:
```
<?php header("Content-type: application/x-javascript"); ?>
window.localStorage.setItem("wt.orskey", "PUT_YOR_KEY_HERE");
<?php echo file_get_contents(__DIR__ . '/config.js');

```

I leave that here for others and suggest to not use it. ;-)
Thanks for helping!
